mws/mws.py

```diff
@@ -248,12 +266,8 @@ def make_request(self, extra_data, method="GET", **kwargs):
         """
         # Remove all keys with an empty value because
         # Amazon's MWS does not allow such a thing.
-        extra_data = remove_empty(extra_data)
 
-        # convert all Python date/time objects to isoformat
-        for key, value in extra_data.items():
-            if isinstance(value, (datetime.datetime, datetime.date)):
-                extra_data[key] = value.isoformat()
+        extra_data = clean_extra_data(extra_data)
```

Comment:
This is called twice at the moment (see other note).

As noted, this feels like the correct place to call the method, but since it is not including `get_default_params()` or `get_proxies()`, it might be missing stuff.

I suggest moving this call to the line just above `if self._test_request_params`. That way our test methods capture every param, all quoted and cleaned up, ready to transmit.

That may require a change to the common test params, as well. We'll just have to try it out.

Reply:
yes thank you so much. I think you nailed it, it's like that.